### ISSUE TITLE
Inconsistent `LRANGE`: using LRANGE it is creating panic and closing the server abruptly

### ISSUE THREAD

**Issue by shashi-sah2003** [DiceDB/dice#1271]
## Steps to reproduce

1. RPUSH mylist "one"
2. RPUSH mylist "two"
3. RPUSH mylist "three"
4. LRANGE mylist -3 2

## Expected output

The expected output when the above set of commands when run on Redis

![image](https://github.com/user-attachments/assets/3f33bdf2-d246-4126-85f4-010168557cbc)

## Observed output

The observed output when the above set of commands when run on DiceDB

![image](https://github.com/user-attachments/assets/2324ad6f-a981-4cbe-be11-d525d150abe4)

Error:
![image](https://github.com/user-attachments/assets/fb134c7e-7b58-466c-a8d4-ee6193e72e30)

The steps to run the test cases are mentioned in the README of the [dice-tests repository](https://github.com/AshwinKul28/dice-tests).

## Expectations for resolution

This issue will be considered resolved when the following things are done

1. changes in the [`dice`](https://github.com/dicedb/dice) code to meet the expected behavior
5. Successful run of the tcl test behavior

You can find the tests under the `tests` directory of the [`dice`](https://github.com/dicedb/dice) repository and the steps to run are in the [README file](https://github.com/dicedb/dice). Refer to the following links to set up DiceDB and Redis 7.2.5 locally

- [setup DiceDB locally](https://github.com/dicedb/dice)
- [setup Redis 7.2.5 locally](https://gist.github.com/arpitbbhayani/94aedf279349303ed7394197976b6843)
- [setup DiceDB CLI](https://github.com/dicedb/dice)

**Comment by shashi-sah2003** [comment]
hey @JyotinderSingh @apoorvyadav1111 @lucifercr07 I would like to work on this, pls assign 
thanks

**Comment by JyotinderSingh** [comment]
> hey @JyotinderSingh @apoorvyadav1111 @lucifercr07 I would like to work on this, pls assign 
> 
> thanks

Assigned

**Comment by srinivasan1995** [comment]
Hey @shashi-sah2003,
just started looking into repo (tried running dicedb locally with dicedb client) and tried replicating issue but couldn't see the issue, is the issue on redis-client side?

checked out dicedb below commit for issue replication (latest master )

```
commit 51ccd048688ca01faaf698f9d850bda50cc875eb (HEAD, origin/master, origin/HEAD)
Author: Shashi sah <86319846+shashi-sah2003@users.noreply.github.com>
Date:   Tue Nov 12 12:32:46 2024 +0545

    #1193: Add type check for string compatibility in evalAPPEND function (#1237)
(END)
```

dicedb-cli commit
```

commit 2c98a6a1b507dd2527720394f37c79d047b94eac (HEAD -> master, origin/master, origin/HEAD)
Author: KaviiSuri <surikavii@gmail.com>
Date:   Mon Nov 4 22:14:30 2024 +0530

    fix: add readable dice colors (#9)
```

```
dicedb> RPUSH mylist "one"
1
dicedb>  RPUSH mylist "two"
2
dicedb> RPUSH mylist "three"
3
dicedb> LRANGE mylist 0 0
1) one
dicedb> LRANGE mylist -3 2
1) one
2) two
3) three
```
